controllers/erpControllers/invoiceController.js

```diff
@@ -155,6 +156,100 @@ methods.update = async (req, res) => {
   }
 };
 
+methods.summary = async (req, res) => {
+  try {
+    let defaultType = 'month';
+    
+    const { type } = req.query;
+
+    if (type) {
+      if (['week', 'month', 'year'].includes(type)) {
+        defaultType = type
+      } else {
+        return res.status(400).json({
+          success: false,
+          result: null,
+          message: 'Invalid type',
+        });
+      }
+    }
+
+    const currentDate = moment();
+    let startDate = currentDate.clone().subtract(1, 'month').startOf('month');
+    let endDate = currentDate.clone().subtract(1, 'month').endOf('month');
+
+    if (defaultType === 'week') {
+      startDate = currentDate.clone().subtract(1, 'week').startOf('week');
+      endDate = currentDate.clone().subtract(1, 'week').endOf('week');
+    }
+    if (defaultType === 'year') {
+      startDate = currentDate.clone().subtract(1, 'year').startOf('year');
+      endDate = currentDate.clone().subtract(1, 'year').endOf('year');
+    }
+
+    const result = await Model.aggregate([
+      {
+        $match: {
+          removed: false,
+          date: {
+            $gte: startDate.toDate(),
+            $lte: endDate.toDate(),
+          },
+        },
+      },
+      {
+        $group: {
+          _id: '$status',
+          count: {
+            $sum: 1,
+          },
+          total_amount: {
+            $sum: '$total',
+          }
+        },
+      },
+      {
+        $project: {
+          _id: 0,
+          status: '$_id',
+          count: 1,
+          percentage: {
+            $multiply: [{ $divide: ['$count', { $sum: '$count' }] }, 100],
```

Comment:
@Ando22 please fix percentage , it doesn't calculate correctly , it always show 100

Reply:
@salahlalami Thank you for the noticing, fixed 🙏